traefik/templates/requirements.yaml

```diff
@@ -8,3 +8,11 @@
 {{- if and .Values.hub.enabled (not (contains "traefik-hub" .Values.image.repository)) }}
   {{- fail "ERROR: traefik-hub image is required when enabling Traefik Hub" -}}
 {{- end }}
+
+{{- if and .Values.providers.kubernetesGateway.enabled (and (semverCompare "<3.1.0-rc3" $version) (not .Values.experimental.kubernetesGateway.enabled)) }}
+  {{- fail "ERROR: Before traefik v3.1.0-rc3, kubernetesGateway is experimental. Enable it by setting experimental.kubernetesGateway.enabled to true" -}}
+{{- end }}
+
+{{- if and .Values.providers.kubernetesGateway.enabled (or (not .Values.providers.kubernetesCRD.enabled) (not .Values.providers.kubernetesIngress.enabled)) }}
+  {{- fail "ERROR: When enabling kubernetesGateway, make sure to enable both kubernetesCRD and kubernetesIngress" -}}
```

Comment:
Mmmh maybe I missed something. Why both providers should be enabled with kubernetesGateway ?

Reply:
You're right :-)